src/main/java/com/synopsys/integration/alert/workflow/startup/AlertStartupInitializer.java

```diff
@@ -67,9 +69,16 @@ public AlertStartupInitializer(final DescriptorMap descriptorMap, final Environm
 
     public void initializeConfigs(final boolean overwriteCurrentConfig) throws IllegalArgumentException, SecurityException, AlertDatabaseConstraintException {
         final Set<String> descriptorNames = descriptorMap.getDescriptorMap().keySet();
+        logger.info("** --------------------------------- **");
+        logger.info("Initializing descriptors with environment variables...");
         for (final String descriptorName : descriptorNames) {
+            logger.info("---------------------------------");
+            logger.info("Descriptor: {}", descriptorName);
+            logger.info("---------------------------------");
             final Map<String, String> newConfiguration = new HashMap<>();
-            final List<DefinedFieldModel> fieldsForDescriptor = descriptorAccessor.getFieldsForDescriptor(descriptorName, ConfigContextEnum.GLOBAL);
+            final List<DefinedFieldModel> fieldsForDescriptor = new ArrayList<>(descriptorAccessor.getFieldsForDescriptor(descriptorName, ConfigContextEnum.GLOBAL));
```

Comment:
Can we create a TreeSet here instead and pass the collection through? If need be, we can modify the DefinedFieldModel to compare itself better (Compare on the key instead of everything)